### PR TITLE
Interruption investigation

### PR DESCRIPTION
**Pull request by r-cyr**

### CODE REVIEW

**Comment by mikearnaldi** [comment]
Reverting the changes to workaround interruption it seems the skipped test now throws an exception somewhere.

Screenshot 2021-09-28 065925

File: packages/system/src/Stream/Stream/effectAsyncInterrupt.ts [hunk 1]
```diff
@@ -17,12 +21,79 @@ import { effectAsyncInterruptEither } from "./effectAsyncInterruptEither"
...
+            toEmit(async (k) => {
+              try {
+                return runtime.runPromise(
+                  pipe(
+                    Take.fromPull(k),
+                    T.chain((_) => Q.offer_(output, _))
+                  )
+                )
+              } catch (e) {
+                if (e instanceof CS.FiberFailure) {
```
Comment (mikearnaldi) [inline packages/system/src/Stream/Stream/effectAsyncInterrupt.ts:50]: better to avoid instanceof comparism

Comment (mikearnaldi) [reply packages/system/src/Stream/Stream/effectAsyncInterrupt.ts:50]: also is there a specific reason to use a promise? runPromise hides the full cause

**Review by mikearnaldi — CHANGES_REQUESTED** [review]

File: packages/system/src/Effect/effectAsyncInterrupt.ts [hunk 2]
```diff
@@ -16,7 +22,7 @@ import { effectMaybeAsyncInterruptBlockingOn } from "./effectMaybeAsyncInterrupt
  * provide better diagnostics.
  */
 export function effectAsyncInterrupt<R, E, A>(
-  register: (cb: Cb<Effect<R, E, A>>) => Canceler<R>,
+  register: (cb: Cb<Effect<R, E, A>>) => E.Either<Canceler<R>, Effect<R, E, A>>,
```
Comment (mikearnaldi) [inline packages/system/src/Effect/effectAsyncInterrupt.ts:25]: didn't we have the *Maybe variants that returned an either?

**Comment by r-cyr** [comment]
Removed Stream effectAsync implementations

**Pull request merged**